pyquil/api/_wavefunction_simulator.py

```diff
@@ -225,58 +244,63 @@ def augment_program_with_memory_values(
 
         return percolate_declares(p)
 
+    def _run_and_measure_request(
```

Comment:
Made these 3 request functions members of `WavefunctionSimulator` and passed gate/measurement noise to request where appropriate.